Code/GraphMol/Substruct/SubstructMatch.cpp

```diff
@@ -236,13 +247,8 @@ namespace RDKit{
     }    
 
 #ifdef RDK_THREADSAFE_SSS
-    if(recursionPossible){
-      ROMol::ConstAtomIterator atIt;
-      for(atIt=query.beginAtoms();atIt!=query.endAtoms();atIt++){
-        if((*atIt)->getQuery()){
-	  detail::ClearSubqueryLocks((*atIt)->getQuery());
-        }
-      }
+    for(size_t i=0;i<locked.size(); i++) {
```

Comment:
not a big deal, but I'd normally do this with either an iterator or BOOST_FOREACH